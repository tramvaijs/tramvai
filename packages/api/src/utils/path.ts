import path from 'node:path';

export const resolveAbsolutePathForFile = ({
  file,
  sourceDir,
  rootDir,
}: {
  file: string;
  sourceDir: string;
  rootDir: string;
}): string => {
  if (path.isAbsolute(file)) {
    return file;
  }
  if (path.isAbsolute(sourceDir)) {
    return path.join(sourceDir, file);
  }
  if (path.isAbsolute(rootDir)) {
    return path.join(rootDir, sourceDir, file);
  }
  return path.resolve(rootDir, sourceDir, file);
};

export const resolveAbsolutePathForFolder = ({
  folder,
  rootDir,
}: {
  folder: string;
  rootDir: string;
}): string => {
  if (path.isAbsolute(folder)) {
    return folder;
  }
  if (path.isAbsolute(rootDir)) {
    return path.join(rootDir, folder);
  }
  return path.resolve(rootDir, folder);
};
