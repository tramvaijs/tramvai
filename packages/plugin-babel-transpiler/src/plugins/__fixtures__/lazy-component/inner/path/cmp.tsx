import React from 'react';

const InnerCmp = () => {
  return <div>inner cmp</div>;
}

export default InnerCmp;
