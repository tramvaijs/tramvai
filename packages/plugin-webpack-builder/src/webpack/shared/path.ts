import path from 'node:path';

export const resolveAbsolutePathForEntry = ({
  entry,
  sourceDir,
  rootDir,
}: {
  entry: string;
  sourceDir: string;
  rootDir: string;
}): string => {
  if (path.isAbsolute(entry)) {
    return entry;
  }
  if (path.isAbsolute(sourceDir)) {
    return path.join(sourceDir, entry);
  }
  if (path.isAbsolute(rootDir)) {
    return path.join(rootDir, sourceDir, entry);
  }
  return path.resolve(rootDir, sourceDir, entry);
};

export const resolveAbsolutePathForOutput = ({
  output,
  rootDir,
}: {
  output: string;
  rootDir: string;
}): string => {
  if (path.isAbsolute(output)) {
    return output;
  }
  if (path.isAbsolute(rootDir)) {
    return path.join(rootDir, output);
  }
  return path.resolve(rootDir, output);
};
