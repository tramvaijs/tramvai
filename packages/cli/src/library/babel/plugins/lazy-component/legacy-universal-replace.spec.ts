import path from 'path';
import pluginTester from 'babel-plugin-tester';
import babelConfig from '../../index';

pluginTester({
  plugin: {}, // плагин уже есть в основном конфиге, если подключить ещё тут то babel упадёт
  pluginName: 'legacy-universal-replace',
  filename: path.join(__dirname, '..', '__fixtures__', 'legacy-universal-replace', 'test'),
  babelOptions: babelConfig({
    typescript: true,
    generateDataQaTag: false,
    loader: false,
    modern: true,
  }),
  tests: {
    'not universal': {
      fixture: 'not-universal.ts',
      snapshot: true,
    },
    'react-universal': {
      fixture: 'react-universal.ts',
      snapshot: true,
    },
    'platform legacy': {
      fixture: 'platform-legacy.ts',
      snapshot: true,
    },
    'tramvai-react': {
      fixture: 'tramvai/react/index.es.ts',
      snapshot: true,
    },
  },
});
