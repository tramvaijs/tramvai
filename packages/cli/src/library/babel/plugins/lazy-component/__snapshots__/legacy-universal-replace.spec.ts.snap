// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`legacy-universal-replace not universal: not universal 1`] = `

const lazy = (imp) => console.log(imp);
const load = lazy(() => import('./cmp'));

(async () => {
  const {default: cmp} = await import('./cmp');

  console.log((cmp as any).actions)
})();

      ↓ ↓ ↓ ↓ ↓ ↓

const lazy = (imp) => console.log(imp);
const load = lazy(() => import('./cmp'));
(async () => {
  const { default: cmp } = await import('./cmp');
  console.log(cmp.actions);
})();


`;

exports[`legacy-universal-replace platform legacy: platform legacy 1`] = `

import asyncUniversal, {asyncUniversalPack} from '@tinkoff/platform-legacy/utils/decorators/asyncUniversal';

asyncUniversal(import('./cmp'))

export const packed = asyncUniversalPack({});

      ↓ ↓ ↓ ↓ ↓ ↓

import asyncUniversal, {
  asyncUniversalPack,
} from '@tinkoff/platform-legacy/utils/decorators/asyncUniversal';
import { lazy as _lazy } from '@tramvai/react';
_lazy({
  chunkName() {
    return 'cmp';
  },
  requireSync(props) {
    return __webpack_require__(this.resolve(props));
  },
  isReady(props) {
    const key = this.resolve(props);
    return !!__webpack_modules__[key];
  },
  importAsync: () => import(/* webpackChunkName: "cmp" */ './cmp'),
  requireAsync(props) {
    const key = this.resolve(props);
    return this.importAsync(props).then((resolved) => {
      return resolved;
    });
  },
  resolve() {
    return require.resolveWeak('./cmp');
  },
});
export const packed = asyncUniversalPack({});


`;

exports[`legacy-universal-replace react-universal: react-universal 1`] = `

import universal from 'react-universal-component';

universal(import('./cmp'))

      ↓ ↓ ↓ ↓ ↓ ↓

import { lazy as _lazy } from '@tramvai/react';
_lazy({
  chunkName() {
    return 'cmp';
  },
  requireSync(props) {
    return __webpack_require__(this.resolve(props));
  },
  isReady(props) {
    const key = this.resolve(props);
    return !!__webpack_modules__[key];
  },
  importAsync: () => import(/* webpackChunkName: "cmp" */ './cmp'),
  requireAsync(props) {
    const key = this.resolve(props);
    return this.importAsync(props).then((resolved) => {
      return resolved;
    });
  },
  resolve() {
    return require.resolveWeak('./cmp');
  },
});


`;

exports[`legacy-universal-replace tramvai-react: tramvai-react 1`] = `

import universal from 'react-universal-component';

export const lazy = (loader) => universal(loader);

      ↓ ↓ ↓ ↓ ↓ ↓

import universal from '@loadable/component';
export const lazy = (loader) => universal(loader);


`;
